### PR TITLE
Add test install errata to one host in a hc

### PR DESCRIPTION
**Pull request by swadeley**
Closed loop: BZ#1810774

[Bug 1810774 - Applying errata via Host Collection the errata are trying to be applied to all hosts associated with the host collection](https://bugzilla.redhat.com/show_bug.cgi?id=1810774)

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/cli/test_errata.py -k test_install_errata_to_one_host
=============================== test session starts ===============================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robottelo
plugins: cov-2.10.0, xdist-1.32.0, services-1.3.1, mock-1.10.4, forked-1.2.0
collecting ... 2020-07-08 15:20:23 - conftest - DEBUG - Collected 34 test cases
collected 34 items / 33 deselected / 1 selected                                                                  

tests/foreman/cli/test_errata.py .                                                                         [100%]

=================================== 1 passed, 33 deselected in 695.11 seconds ====================

File: tests/foreman/cli/test_errata.py [hunk 1]
```diff
@@ -495,6 +495,52 @@ def test_positive_list_affected_chosts(self):
...
+        the host collection
+
+
+        :Setup: Errata synced on satellite server.
+
+        :Steps:
+           1. Remove FAKE_2_CUSTOM_PACKAGE_NAME packages from one host
+           2. host-collection erratum install --errata <errata> --id <id>
+            --organization <org name>
+           3. Assert first host does not have any FAKE_2_CUSTOM_PACKAGE_NAME packages.
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_errata.py:514]: `packages./package./g`

**Comment by swadeley** [comment]
Hello @latran , you are secondary for this test's component, please read & review if possible.

Thank you

File: tests/foreman/cli/test_errata.py [hunk 2]
```diff
@@ -495,6 +495,52 @@ def test_positive_list_affected_chosts(self):
...
+           4. Assert second host does have FAKE_2_CUSTOM_PACKAGE
+
+        :expectedresults: Erratum is only installed on one host.
+
+        :BZ: 1810774
+        """
+        # Remove package on first VM to remove need for CUSTOM_ERRATA_ID
+        result = ssh.command(
+            f'yum erase -y {self.CUSTOM_PACKAGE}', self.virtual_machines[0].ip_addr
+        )
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:524]: `VirtualMachine` objects provide `run` method exactly so you don't have to extract ip address of machine manually.

```suggestion
        result = self.virtual_machines[0].run(f'yum erase -y {self.CUSTOM_PACKAGE}')
```

File: tests/foreman/cli/test_errata.py [hunk 3]
```diff
@@ -495,6 +495,52 @@ def test_positive_list_affected_chosts(self):
...
+            {
+                'id': self.host_collection['id'],
+                'organization': self.org['name'],
+                'errata': [self.CUSTOM_ERRATA_ID],
+            }
+        )
+        Task.progress({'id': install_task[0]['id']})
+        # Assert first host does not have any FAKE_1_CUSTOM_PACKAGE_NAME packages
+        result = ssh.command(
+            f'rpm -q {FAKE_1_CUSTOM_PACKAGE_NAME}', self.virtual_machines[0].ip_addr
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:537]: Same comment about `VirtualMachine.run()` as above.

File: tests/foreman/cli/test_errata.py [hunk 4]
```diff
@@ -495,6 +495,52 @@ def test_positive_list_affected_chosts(self):
...
+        )
+        assert result.return_code == 1, "Unwanted custom package found."
+        # Assert second host does have FAKE_2_CUSTOM_PACKAGE
+        result = ssh.command(f'rpm -q {FAKE_2_CUSTOM_PACKAGE}', self.virtual_machines[1].ip_addr)
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:541]: And here.

File: tests/foreman/cli/test_errata.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:524]: OK, change made, thank you

File: tests/foreman/cli/test_errata.py (hunk 4 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:541]: OK, change made, thank you

File: tests/foreman/cli/test_errata.py (hunk 3 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:537]: OK, change made, thank you

**Comment by swadeley** [comment]
robottelo]$ pytest tests/foreman/cli/test_errata.py -k test_install_errata_to_one_host
================================== test session starts ====================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robottelo
plugins: cov-2.10.0, xdist-1.32.0, services-1.3.1, mock-1.10.4, forked-1.2.0
collecting ... 2020-07-14 08:07:07 - conftest - DEBUG - Collected 34 test cases
collected 34 items / 33 deselected / 1 selected                                                                  

tests/foreman/cli/test_errata.py .                                                                         [100%]

============================= 1 passed, 33 deselected in 707.58 seconds =================

**Comment by swadeley** [comment]
robottelo]$ pytest tests/foreman/cli/test_errata.py -k test_install_errata_to_one_host
================================= test session starts ===============================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robottelo
plugins: cov-2.10.0, xdist-1.32.0, services-1.3.1, mock-1.10.4, forked-1.2.0
collecting ... 2020-07-14 08:27:41 - conftest - DEBUG - Collected 34 test cases
collected 34 items / 33 deselected / 1 selected                                                                  

tests/foreman/cli/test_errata.py .                                                                         [100%]

=========================== 1 passed, 33 deselected in 704.15 seconds ===================

File: tests/foreman/cli/test_errata.py [hunk 5]
```diff
@@ -495,6 +495,48 @@ def test_positive_list_affected_chosts(self):
...
             ), "VM host name not found in list of applicable hosts"
 
+    @tier3
+    def test_install_errata_to_one_host(self):
+        """Install an erratum to one of the hosts in a host collection.
+
+        :id: bfcee2de-3448-497e-a696-fcd30cea9d33
+
+        :expectedresults: Errata was successfully installed in only one of the hosts in
+        the host collection
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:505]: Travis complains about this line, says it should be indented.
When trying to fix, you can `make test-docstrings` in your branch to get feedback faster than pushing and waiting for travis.

Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: @mirekdlugosz thanks for the tip, `make test-docstrings` found one other error.

**Comment by swadeley** [comment]
robottelo]$ pytest tests/foreman/cli/test_errata.py -k test_install_errata_to_one_host
=================================== test session starts ==========================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robottelo
plugins: cov-2.10.0, xdist-1.32.0, services-1.3.1, mock-1.10.4, forked-1.2.0
collecting ... 2020-07-14 11:28:22 - conftest - DEBUG - Collected 34 test cases
collected 34 items / 33 deselected / 1 selected                                                                  

tests/foreman/cli/test_errata.py .                                                                         [100%]

=================================== 1 passed, 33 deselected in 643.26 seconds ==============

**Comment by swadeley** [comment]
Thank you

**Pull request merged**